daemon/cmd/daemon_main.go

```diff
@@ -1458,6 +1462,12 @@ func (d *Daemon) instantiateAPI() *restapi.CiliumAPI {
 	return restAPI
 }
 
+func initHealthCheckNodePortOption() {
```

Comment:
I think it would make more sense to have this inside `initKubeProxyReplacementOptions` so we don't forget to update the logic if we ever change the replacement modes.

Reply:
Agreed. I thought about it, then went for a new function.